format/mbr/mbr.go

```diff
@@ -0,0 +1,72 @@
+package mbr
+
+import (
+	"embed"
+	"fmt"
+
+	"github.com/wader/fq/format"
+	"github.com/wader/fq/format/registry"
+	"github.com/wader/fq/pkg/decode"
+	"github.com/wader/fq/pkg/scalar"
+)
+
+//go:embed mbr.jq
+var mbrFS embed.FS
+
+func init() {
+	registry.MustRegister(decode.Format{
+		Name:        format.MBR,
+		Description: "Master Boot Record",
+		DecodeFn:    mbrDecode,
+		Functions:   []string{"_help"},
+		Files:       mbrFS,
+	})
+}
+
+func decodePartitionTableEntry(d *decode.D) {
+	d.FieldU8("boot_indicator", scalar.UToDescription{
+		0x80: "active",
+		0x00: "inactive",
+	})
+	d.FieldStrScalarFn("starting_chs_vals", decodeCHSBytes)
+	d.FieldU8("partition_type", partitionTypes)
+	d.FieldStrScalarFn("ending_chs_vals", decodeCHSBytes)
+	d.FieldStrScalarFn("starting_sector", decodeCHSBytes)
+	d.U8() // extra byte
+	d.FieldScalarU32("partition_size")
+}
+
+// Because this is a fixed-sized table, I am opting to use a
+// FieldStruct instead of a FieldArray
+func decodePartitionTable(d *decode.D) {
+	d.FieldStruct("entry_1", decodePartitionTableEntry)
+	d.FieldStruct("entry_2", decodePartitionTableEntry)
+	d.FieldStruct("entry_3", decodePartitionTableEntry)
+	d.FieldStruct("entry_4", decodePartitionTableEntry)
+}
+
+// Source: https://thestarman.pcministry.com/asm/mbr/PartTables.htm#Decoding
+func decodeCHSBytes(d *decode.D) scalar.S {
+	head, _ := d.Bits(8)
+	sectorHighBits, err := d.Bits(2)
+	if err != nil {
+		d.IOPanic(err, "chs")
+	}
+	sector, _ := d.Bits(6)
+	cylinderLowerBits, err := d.Bits(8)
+	if err != nil {
+		d.IOPanic(err, "chs")
+	}
+	cylinder := (sectorHighBits << 2) | cylinderLowerBits
```

Comment:
Oh the cylinder bits are "continuous" as 16LE, this is a bit messy to represent :( i've run into this issue with some other formats like https://github.com/wader/fq/blob/master/format/vpx/vp8_frame.go#L36 haven't figured any nice way to handle this yet. Also would be nice if there was some decoding helpers for doing this